### ISSUE TITLE
SQL Error when import computer with softwares

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#501]
---

Author Name: **David Durieux** (@ddurieux)
Original Redmine Issue: 999, http://forge.fusioninventory.org/issues/999
Original Date: 2011-07-02
Original Assignee: David Durieux

---

```
*** MySQL query error :
***
SQL: INSERT
                   INTO `glpi_computers_softwareversions` (`computers_id`,`softwareversions_id`,`is_template`,`is_deleted`) VALUES (\'2656\',\'20845\',\'0\',\'0\')
Error: Duplicate entry '2656-20845' for key 'unicity'
Backtrace :
/Donnees/www/data-dist/glpi080/inc/commondbtm.class.php&nbsp;:355               DBmysql->query()
/Donnees/www/data-dist/glpi080/inc/commondbtm.class.php&nbsp;:665               CommonDBTM->addToDB()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/import_software.class.php&nbsp;:123          CommonDBTM->add()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/libhook.class.php&nbsp;:406          PluginFusinvinventoryImport_Software->addSoftware()
&nbsp;:         PluginFusinvinventoryLibhook::addSections()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/lib.class.php&nbsp;:578              call_user_func()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/lib.class.php&nbsp;:163              PluginFusinvinventoryLib->updateLibMachine()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/inventory.class.php&nbsp;:220                PluginFusinvinventoryLib->startAction()
/Donnees/www/data-dist/glpi080/plugins/fusioninventory/inc/ruleimportequipment.class.php&nbsp;:578              PluginFusinvinventoryInventory->rulepassed()
/Donnees/www/data-dist/glpi080/inc/rule.class.php&nbsp;:770             PluginFusioninventoryRuleImportEquipment->executeActions()
/Donnees/www/data-dist/glpi080/inc/rulecollection.class.php&nbsp;:668           Rule->process()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/inventory.class.php&nbsp;:139                RuleCollection->processAllRules()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/inc/importxml.class.php&nbsp;:89         PluginFusinvinventoryInventory->sendCriteria()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/front/importxml.php&nbsp;:52             PluginFusinvinventoryImportXML->importXMLFile()
/Donnees/www/data-dist/glpi080/plugins/fusinvinventory/front/importxml.php

```

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Durieux** (@ddurieux)
Original Date: 2011-07-02T20:36:20Z

---

This is the problem when have 2 same softwares, bug of agent fixed #927 but have this error with old agent version.
So in this case check if this software is yet here, if yes refuse to import it.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Durieux** (@ddurieux)
Original Date: 2011-07-02T20:43:08Z

---

Applied in changeset commit:ce05618f6ed22e8723533a0738ffddc1b6eeab7c.
